app/webpacker/components/RegistrationsV2/RegistrationAdministration/RegistrationAdministrationList.jsx

```diff
@@ -89,6 +90,20 @@ export default function RegistrationAdministrationList({ competitionInfo }) {
     },
   });
 
+  const { mutate: bulkAutoAcceptMutation, isPending: isAutoAccepting } = useMutation({
+    mutationFn: bulkAutoAccept,
+    onError: () => {
+      dispatchStore(showMessage(
+        'competitions.registration_v2.auto_accept.cant_bulk_auto_accept',
+        'negative',
+      ));
+    },
+    onSuccess: async () => {
+      dispatchStore(showMessage('competitions.registration_v2.auto_accept.bulk_auto_accepted', 'positive'));
+      await refetch();
+    },
```

Comment:
Why `async`/`await` here? I know that `refetch` itself is an async method, but we don't get any benefit from waiting for the result (afaik)

Reply:
This might be my JS inexperience - in my mind if refetch() is async then we have to use async/await? 

My desired outcome is the following: 
- refetch the registrations after 
- have the various registration list components be `loading` until we're finished fetching the registrations - so `await`ing the results does naively seem like something I want to do?

Reply:
Aha ok that makes sense, thanks for the explanation! 

I have tried both, and I'm keeping async/await because without them the UI re-enables _slightly_ before the `refetch` has reflected in the registration tables - ie, they enable and then the registrations move to their new list. With async/await, when the tables re-enable (ie exit the loading state) the registrations are already in their correct/new list.

Reply:
That... Shouldn't happen :sweat_smile: The "outermost" call to your mutation isn't blocking, so forcing the inner `onSuccess` hook to block and wait should have zero influence on the outside world code flow. Let me try and investigate.

Reply:
Aha, very interesting! Turns out this is tied to the internal React rendering cycle: https://tkdodo.eu/blog/mastering-mutations-in-react-query#common-gotchas and the section on `onSuccess` at https://tanstack.com/query/latest/docs/framework/react/reference/useMutation are an interesting read.

Reply:
Probably the best compromise here is to _return_ the `Promise` generated by calling `refetch`. That way, you don't have to care about `await`ing the result yourself, and thus you don't have to mark your method as `async`. The mutation's inner workings will take care of "cleanly" resolving the value for you: https://tkdodo.eu/blog/mastering-mutations-in-react-query#awaited-promises

Reply:
Ah yep adding a `return` makes it work as expected!